README.md

```diff
@@ -1,38 +1,64 @@
-  This includes in file names (for example, rename directory
-  `src/main/java/org/checkerframework/checker/templatefora/`).
-* Change the groupId of the `publishing` block in the `build.gradle` file
-  to an organization you belong to that can publish to Maven
-  Central. Change the name of the package containing the source (currently
-  `org.checkerframework.checker.templatefora`) so that it reflects your
-  organization's naming standards.
-* Change the copyright info in the `LICENSE` file.  You may change the license as well, if you wish.
-* Modify this README so that it describes your checker, instead of this text. :)
-* Follow the ["How to create a new checker" instructions](https://checkerframework.org/manual/#creating-a-checker) in the Checker Framework Manual .
-
-### How the build system works
-
-This template uses [Gradle](https://gradle.org/) as its build system.
-To build the checker, run `./gradlew build` from the top-level directory.
-
-The `build.gradle` file contains the definition of the build. The only
-significant modification from a standard Gradle Java project is the
-addition of the Maven publishing plugin, which permits you to publish
-your checker to your local Maven repository by running `./gradlew publish`.
-This is useful for testing out your checker on a real codebase before you
-publish it elsewhere, such as to Maven Central.
+### No Literal Checker
+
+A typechecker for proving that no literal values can flow to specified APIs in
+a Java codebase. Useful for enforcing security and compliance rules like 
+"do not hard-code credentials".
+
+### What problem does this solve?
+
+Many interesting security/compliance properties are requirements that the value
+passed to a particular API never be a constant. For example, cryptographic keys 
+and passwords need to come from user configuration or computation, not from a 
+hard-coded constant.
+
+### How does it work?
+
+The checker is built on the [Checker Framework](checkerframework.org).
+
+The type system is:
+
+```java
+@MaybeConstant
+      |
+ @NonConstant
+```
+
+The default is `@NonConstant` for all expressions except manifest literals (examples
+of manifest literals include `1`, `"hello"`, and `{0xa, 0xb}`). That default includes
+method parameters, so a user should expect to need to write an annotation on the defintion
+of any method that is called with a constant argument. For example, if your code includes
+the call:
+```java
+foo(5);
+```
+then the defintion of `foo` will need to be annotated like so:
+```java
+void foo(@MaybeConstant int x) { }
+```
+
+The default is slightly different in unchecked code (that is, code for which
+only the bytecode is available, such as code from a `.jar` file). The following
+optimistic defaulting rules are applied:
+* the return type of any function defined in a library is `@NonConstant`
+* the types of a libraries' formal parameters are always `@MaybeConstant` unless
```

Comment:
You tend to use odd rules for possessives.

To form a possessive:
 * Use 's if the noun is singular or is a plural that does not end in "s".
 * Use ' if the noun is plural *and* ends in "s".

This one should be "library's".